assets/pages/new-contributions-landing/components/SetPassword/SetPassword.jsx

```diff
@@ -0,0 +1,153 @@
+
+import React from 'react';
+import { connect } from 'react-redux';
+import { type Dispatch } from 'redux';
+
+import { type Contrib } from 'helpers/contributions';
+import { classNameWithModifiers } from 'helpers/utilities';
+import { setPasswordGuest } from 'helpers/paymentIntegrations/readerRevenueApis';
+import type { Csrf as CsrfState } from 'helpers/csrf/csrfReducer';
+import SvgPasswordKey from 'components/svgs/passwordKey';
+import SvgEnvelope from 'components/svgs/envelope';
+
+import { NewContributionTextInput } from '../ContributionTextInput';
+import { type ThankYouPageStage } from '../../contributionsLandingReducer';
+import { setThankYouPageStage, setPasswordHasBeenSubmitted, updatePassword, type Action } from '../../contributionsLandingActions';
+import { checkEmail, emailRegexPattern } from '../../formValidation';
+import { ButtonWithRightArrow } from '../ButtonWithRightArrow';
+
+// ----- Types ----- //
+
+/* eslint-disable react/no-unused-prop-types */
+type PropTypes = {
+  contributionType: Contrib,
+  email: string,
+  password: string,
+  guestAccountCreationToken: string,
+  setThankYouPageStage: (ThankYouPageStage) => void,
+  setPasswordHasBeenSubmitted: () => void,
+  passwordHasBeenSubmitted: boolean,
+  updatePassword: (Event) => void,
+  csrf: CsrfState,
+};
+/* eslint-enable react/no-unused-prop-types */
+
+
+// ----- State Maps ----- //
+
+const mapStateToProps = state => ({
+  contributionType: state.page.form.contributionType,
+  email: state.page.form.formData.email,
+  password: state.page.form.setPasswordData.password,
+  passwordHasBeenSubmitted: state.page.form.setPasswordData.passwordHasBeenSubmitted,
+  guestAccountCreationToken: state.page.form.guestAccountCreationToken,
+  csrf: state.page.csrf,
+});
+
+function mapDispatchToProps(dispatch: Dispatch<Action>) {
+  return {
+    setThankYouPageStage: (thankYouPageStage: ThankYouPageStage) => {
+      dispatch(setThankYouPageStage(thankYouPageStage));
+    },
+    setPasswordHasBeenSubmitted: () => {
+      dispatch(setPasswordHasBeenSubmitted());
+    },
+    updatePassword: (event: Event) => {
+      if (event.target instanceof HTMLInputElement) {
+        dispatch(updatePassword(event.target.value));
+      }
+    },
+  };
+}
+
+
+// ----- Functions ----- //
+
+function onSubmit(props: PropTypes): Event => void {
+  return (event) => {
+    props.setPasswordHasBeenSubmitted();
+    event.preventDefault();
+
+    if (!(event.target: any).checkValidity()) {
+      return;
+    }
+
+    // TODO: send user to thank you page with error if password was not set
+    setPasswordGuest(props.password, props.guestAccountCreationToken, props.csrf)
+      .then((response) => {
+        if (response === true) {
+          props.setThankYouPageStage('thankYouPasswordSet');
+        } else {
+          props.setThankYouPageStage('thankYouPasswordNotSet');
+        }
+      });
+  };
+}
+
+
+// ----- Render ----- //
+
+function SetPassword(props: PropTypes) {
```

Comment:
Should this whole module bubble up to the top of the application, next to the form and thank you page in [here](https://github.com/guardian/support-frontend/blob/master/assets/pages/new-contributions-landing/contributionsLanding.jsx#L56)?